frontend/src/app/main/ui/workspace/viewport/outline.cljs

```diff
@@ -92,18 +92,17 @@
         edition     (obj/get props "edition")
         zoom        (obj/get props "zoom")
 
-        outlines    (set/union selected hover)
-
-        shapes      (d/concat-set
-                     (->> outlines
-                          (filter #(not= edition %))
-                          (map #(get objects %))
-                          (filterv show-outline?)
-                          (filter some?))
-                     ;; outline highlighted shapes even if they are hidden or blocked
-                     (->> highlighted
-                          (filter #(not= edition %))
-                          (map #(get objects %))
-                          (filter some?)))]
+        lookup      (d/getf objects)
+        edition?    (fn [o] (= edition o))
+
+        shapes      (-> #{}
+                        (into (comp (remove edition?)
+                                    (keep lookup)
```

Comment:
Wow, I didn't know this fn existed
I like how it spares the need to `(remove nil?)` after a `(map lookup)`, more concise